### DISCUSSION TITLE
Syntax has much room for improvement

### DISCUSSION THREAD

**Discussion by Jakobovski** [carbon-language/carbon-lang#1690] [Language design]
First of all awesome project! Great work, a much needed language.

I think the syntax has much room for improvement. It's readability is mediocre when it should be great. As a simple example i32 imo should be int32, it's just way more readable. A beautiful, readable, and more standard syntax makes a language far more pleasant to use, encourages adoptions, and will result in less buggy code

If you think I am wrong and the syntax is good, I suggest you do a poll on twitter and see what people think.

If it's not too late to improve the syntax I would be happy to help out and come up with a good a design.

Cross posted: https://github.com/carbon-language/carbon-lang/issues/1686

**Comment by OlaFosheimGrostad** [comment]
Yes, fortunately syntax is easy to change later, so semantics are more important at this stage I think?

Anyway, polls don't provide more than a focus group do, you basically learn what might prevent people from downloading the language and give it a spin, but you don't learn anything about usability. So that is more akin to market-research, people will tend to front what they are familiar with or reproduce assumptions and popular opinions (like what is fashionable). Voting on designs tend to lead to bad mishmash solutions, so I hope they don't do that…

Most unfortunately, both designers and users are not very good at evaluating usability. Designers tend to get blind to their design. This is universal, there is a reason why mastering in music production is done independently, game developers fail experience their games as game players do, artists have to set their work aside for a few months to detach themselves from it and so on.

Which is why usability evaluation *should* be done by external usability experts that has had nothing to do with the design process, and not by the designers themselves or end users. But the real usability evaluation cannot be done until you sit down with a representative set of programmers (different backgrounds and programming domains) and do real evaluation of actual use in a realistic setting (e.g. by thinking aloud protocols.).

But I don't know what role this projects plays in Google, so… that might be too costly?

**Comment by lexi-nadia** [comment]
Carbon is still new, and almost nothing is set in stone. That said, one of the project's top goals is that Carbon code should be easy to understand, read, and write: https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write

Specific proposals for improving syntax are always welcome, especially if they come with motivating examples that provide new information.
